### ISSUE TITLE
Getting all nodes are offline message

### ISSUE THREAD

**Issue by vasuems** [jenkinsci/docker-plugin#431]
I am using docker container 1.11, jenkins 2.7.2  and docker plugin 0.16.1. I am getting all nodes are offline message. I tried the below.
1. Create ssh keys for root user and use that as the authentication key in the jenkins configuration
2. Create ssh keys for jenkins user and use that as the authentication key in the jenkins configuration

I tried the steps mentioned in http://varmenise.tumblr.com/post/87976165063/docker-on-centos-jenkins. 

Still getting all nodes are offline. 

Previously I have used docker 1.12 on RHEL. That time many nodes have been created. Later I used ubuntu with docker version 1.11.

**Comment by ndeloof** [comment]
Please upgrade to 0.17+ which inject ssh keys to enforce a successful authentication. If still not working, please check created agent container and capture their logs
(`docker ps -a` to see all containers including those failing to start / exited, `docker logs xxx` to collect logs)

**Comment by ndeloof** [comment]
please re-open this issue if this is still relevant on 1.0.2 and later

**Comment by vamsidhar5434** [comment]
I am also facing the same issues all nodes creates were offline

**Comment by jkiddo** [comment]
Same to me ...

**Comment by vamsidhar5434** [comment]
surprisingly it starting working after enabling the security for jenkins

**Comment by leonardocastanodiaz** [comment]
Hello guys, somebody can share how this issue was solved?

@vamsidhar5434 apparently you have enabled some security for Jenkins, can you share what you have done?

Many thanks

**Comment by pjdarton** [comment]
A common cause I've seen where I work is the SSH key verification setting (which was introduced some time ago now, but if you've only recently upgraded your ssh-slaves plugin then you'll've only recently met this).
The SSH plugin _can_ verify that the SSH host key of the slave it's connecting to is "known" to Jenkins, and it tends to default to that in an upgrade situation (where the user hadn't previously set it).  For static slaves this is, obviously, a good idea from a security point of view as it prevents a man-in-the-middle attack.
For docker usage, every new docker container tends to have its own unique key that isn't "known" to Jenkins, so it's important that the "SSH key verification strategy" is set to "non-verifying", otherwise the SSH plugin will decline the connection and the node will remain offline.

TL;DR: Manage Jenkins  -> Configure System, scroll down to the docker cloud, expand the template(s), look for the connection method and, if it's SSH, make sure that the key verification strategy is non-verifying.
